### ISSUE TITLE
NavigationView onDestroy() throws  java.lang.IllegalArgumentException: Service not registered: android.speech.tts.TextToSpeech$Connection

### ISSUE THREAD

**Issue by pwojnar** [mapbox/mapbox-navigation-android#2438]
**Android API: 27 and 28**
**Mapbox Navigation SDK version: v0.42.4**

### Steps to trigger behavior

 1. I include a NavigationView in my activity. When the activity is initialized, I call onCreate(), initialize() and then onStart() and onResume() on it
 2. In the onNavigationReady() callback I call:
```
val speechPlayer = NavigationSpeechPlayer(SpeechPlayerProvider(this, getApplicationLanguage().toLanguageTag(), true,
                VoiceInstructionLoader(this, BuildConfig.mapBoxKey,
                        Cache(File(application.cacheDir,  SPEECH_CACHE_NAME), SPEECH_CACHE_SIZE))))
navigationView.startNavigation(NavigationViewOptions.builder()
                    .speechPlayer(speechPlayer)
                    .directionsRoute(route)
                    .build())
```
3. In my activity's onDestroy() method, I call navigationView.onDestroy()

### Expected behavior
The navigationView will be destroyed without exception

### Actual behavior
In many cases, on various phones and Android API levels, I get the following exception, when the user leaves the activity:
```
Fatal Exception: java.lang.RuntimeException: Unable to destroy activity {my.app/my.app.navigation.CarNavigationActivity}: java.lang.IllegalArgumentException: Service not registered: android.speech.tts.TextToSpeech$Connection@3aafce9
       at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:5176)
       at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:5195)
       at android.app.servertransaction.DestroyActivityItem.execute(DestroyActivityItem.java:39)
       at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:145)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2199)
       at android.os.Handler.dispatchMessage(Handler.java:112)
       at android.os.Looper.loop(Looper.java:216)
       at android.app.ActivityThread.main(ActivityThread.java:7625)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:524)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:987)
Caused by java.lang.IllegalArgumentException: Service not registered: android.speech.tts.TextToSpeech$Connection@3aafce9
       at android.app.LoadedApk.forgetServiceDispatcher(LoadedApk.java:1755)
       at android.app.ContextImpl.unbindService(ContextImpl.java:1925)
       at android.content.ContextWrapper.unbindService(ContextWrapper.java:717)
       at android.speech.tts.TextToSpeech$Connection.disconnect(TextToSpeech.java:2294)
       at android.speech.tts.TextToSpeech$1.run(TextToSpeech.java:862)
       at android.speech.tts.TextToSpeech$1.run(TextToSpeech.java:857)
       at android.speech.tts.TextToSpeech$Connection.runAction(TextToSpeech.java:2314)
       at android.speech.tts.TextToSpeech.runAction(TextToSpeech.java:759)
       at android.speech.tts.TextToSpeech.runActionNoReconnect(TextToSpeech.java:745)
       at android.speech.tts.TextToSpeech.shutdown(TextToSpeech.java:857)
       at com.mapbox.services.android.navigation.ui.v5.voice.AndroidSpeechPlayer.onDestroy(AndroidSpeechPlayer.java:112)
       at com.mapbox.services.android.navigation.ui.v5.voice.SpeechPlayerProvider.onDestroy(SpeechPlayerProvider.java:82)
       at com.mapbox.services.android.navigation.ui.v5.voice.NavigationSpeechPlayer.onDestroy(NavigationSpeechPlayer.java:86)
       at com.mapbox.services.android.navigation.ui.v5.NavigationViewModel.deactivateInstructionPlayer(NavigationViewModel.java:499)
       at com.mapbox.services.android.navigation.ui.v5.NavigationViewModel.onDestroy(NavigationViewModel.java:138)
       at com.mapbox.services.android.navigation.ui.v5.NavigationView.shutdown(NavigationView.java:730)
       at com.mapbox.services.android.navigation.ui.v5.NavigationView.onDestroy(NavigationView.java:188)
       at my.app.navigation.CarNavigationActivity.onDestroy(CarNavigationActivity.kt:194)
       at android.app.Activity.performDestroy(Activity.java:7737)
       at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1327)
       at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:5161)
       at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:5195)
       at android.app.servertransaction.DestroyActivityItem.execute(DestroyActivityItem.java:39)
       at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:145)
       at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2199)
       at android.os.Handler.dispatchMessage(Handler.java:112)
       at android.os.Looper.loop(Looper.java:216)
       at android.app.ActivityThread.main(ActivityThread.java:7625)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:524)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:987)
```
After that, my app crashes

**Comment by Guardiola31337** [comment]
cc @JunDai

**Comment by Guardiola31337** [comment]
@JunDai could we check if this would be an issue in `1.0`? Probably fixed by #3297, right?

**Comment by cafesilencio** [comment]
This issue cannot be reproduced in 1.0. Please be on the lookout for our first public preview of UI SDK v1.0 coming soon. Feel free to create a new issue if you feel it still exists.
